posts/batchcorrection/index.qmd

```diff
@@ -0,0 +1,736 @@
+
+
+### Get overlapping genes (all the genes in this case) to combine the two datasets
+overlapping_genes <- intersect(rownames(sem_v20), rownames(sem_v15))
+combined_metadata <- data.table::rbindlist(list(
+  sem_v20@meta.data
+  ,sem_v15@meta.data
+  )
+  ,use.names=TRUE,fill = TRUE)
+combined_metadata[cell_ID %in% colnames(sem_v15),batchid:="v1.5"]
+combined_metadata[cell_ID %in% colnames(sem_v20),batchid:="v2.0"]
+
+### Create a combined seurat object
+sem_comb <- 
+Seurat::CreateSeuratObject(counts = cbind(sem_v20[["RNA"]]@counts[overlapping_genes,]
+                                          ,sem_v15[["RNA"]]@counts[overlapping_genes,])
+                           ,meta.data = data.frame(combined_metadata
+                                                   ,row.names = combined_metadata$cell_ID)
+                           )
+
+### offset the x-coordinate of v1.5 to 
+### more easily plot both tissues side-by-side
+x_offset <- max(sem_comb@meta.data$x_slide_mm[sem_comb@meta.data$batchid=="v2.0"]) + 1
+sem_comb@meta.data$x_slide_mm[sem_comb@meta.data$batchid=="v1.5"] <-  sem_comb@meta.data$x_slide_mm[sem_comb@meta.data$batchid=="v1.5"]  + x_offset
+  
+```
+
+
+
+# 'Harmony Analysis' {#tldr}
+
+With our combined dataset containing all batches, we'll perform a few standard tertiary analysis steps using the `Seurat` package.
+After principal component analysis, we will pass the PC embeddings to Harmony for batch correction.
+
+```{r}
+#| eval: false 
+#| echo: true
+
+### Normalize, Scale, and Run PCA on the combined dataset 
+#### (standard data analysis steps so far)
+sem_comb <- Seurat::FindVariableFeatures(sem_comb)
+normed <- Seurat::LogNormalize(sem_comb[["RNA"]]@counts
+                                 ,scale.factor = mean(Matrix::colSums(sem_comb[["RNA"]]@counts))
+                                 )
+sem_comb <- Seurat::SetAssayData(sem_comb, layer="data", new.data = normed)
+sem_comb <- Seurat::ScaleData(sem_comb)
+sem_comb <- Seurat::RunPCA(sem_comb)
+
+
+```
+
+```{r}
+#| eval: false
+#| echo: false
+
+saveRDS(pcaobj, "figures/pcaobj.combined.regular.rds")
+
+```
+
+```{r}
+#| eval: false
+#| echo: false
+
+pcaobj <- readRDS("figures/pcaobj.combined.regular.rds")
+sem_comb[["pca"]] <- pcaobj
+
+```
+
+Now we run harmony with our PCA embeddings, specifying `"batchid"` as our variable to adjust for (our indicator for v1.5 and v2.0).
+```{r}
+#| eval: false
+#| echo: true
+
+harmony_embeddings <- harmony::RunHarmony(sem_comb[["pca"]]@cell.embeddings
+                                          ,meta_data = sem_comb@meta.data
+                                          ,vars_use = "batchid"
+                                          )
+
+```
+
+```{r}
+#| eval: false
+#| echo: false
+
+
+saveRDS(harmony_embeddings, "figures/harmony_embeddings.rds")
+
+```
+```{r}
+#| eval: false
+#| echo: false
+
+
+harmony_embeddings <- readRDS("figures/harmony_embeddings.rds")
+harmony_embeddings <- readRDS("figures/harmony_embeddings.rds")
+
```

Comment:
Add something showing the structure of the harmony embeddings, e.g. harmony_embeddings[1:5,1:5] or str(harmony_embeddings).